services/src/main/java/org/keycloak/authentication/forms/RegistrationFormsHelper.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2023 Red Hat, Inc. and/or its affiliates
+ *  and other contributors as indicated by the @author tags.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+
+package org.keycloak.authentication.forms;
+
+import jakarta.ws.rs.core.MultivaluedHashMap;
+import jakarta.ws.rs.core.MultivaluedMap;
+import org.keycloak.authentication.FormContext;
+import org.keycloak.models.KeycloakSession;
+import org.keycloak.userprofile.UserProfile;
+import org.keycloak.userprofile.UserProfileContext;
+import org.keycloak.userprofile.UserProfileProvider;
+
+/**
+ * @author <a href="mailto:mposolda@redhat.com">Marek Posolda</a>
+ */
+public class RegistrationFormsHelper {
```

Comment:
I'm not 100% sure if we should filter attributes outside the user profile. The idea is to actually centralize how they are handled.

As you mentioned, we are only accepting custom attributes from being recognized by the user profile engine. None of our attributes should be using the `user.attributes` prefix. There is a test now covering this behavior too.

But if you really want to blacklist those password-related attributes, I think we should do it on the provider side.

Wdyt?

Reply:
Yeah, that method is the best place ...

Reply:
Ok, will update the PR in that regards.

Reply:
Actually ... I remember now what we discussed with Stian. It should be better to keep your current approach and not couple user profile providers with other components.